src/tools/schema2kotlin.ts

```diff
@@ -89,36 +95,42 @@ class KotlinGenerator implements ClassGenerator {
       typeDecls = '\n' + aliases.map(a => `typealias ${a} = ${name}`).join('\n') + '\n';
     }
 
+    const withFields = (populate: string) => fieldCount === 0 ? '' : populate;
+    const withoutFields = (populate: string) => fieldCount === 0 ? populate : '';
+
     return `\
 
-data class ${name}(
-  ${this.fields.join(', ')}
-) : Entity<${name}>() {
-  override fun decodeEntity(encoded: String): ${name}? {
-    if (encoded.isEmpty()) {
-      return null
-    }
-    val decoder = StringDecoder(encoded)
-    this.internalId = decoder.decodeText()
-    decoder.validate("|")
-    var i = 0
-    while (!decoder.done() && i < ${fieldCount}) {
-      val name = decoder.upTo(":")
-      when (name) {
-        ${this.decode.join('\n        ')}
-      }
-      decoder.validate("|")
-      i++
+${withFields('data ')}class ${name}(${ withFields(`\n    ${this.fields.join(',\n    ')}\n`) }) : Entity<${name}>() {
+
+    override fun decodeEntity(encoded: String): ${name}? {
+        if (encoded.isEmpty()) {
+            return null
+        }
+        val decoder = StringDecoder(encoded)
+        this.internalId = decoder.decodeText()
```

Comment:
You don't need to use `this.` unless there is a local variable overshadowing a class's field.